src/reconcile/rec_visibility.c

```diff
@@ -97,8 +91,6 @@ __rec_append_orig_value(
     total_size = size = 0;     /* -Wconditional-uninitialized */
     if (unpack == NULL || unpack->type == WT_CELL_DEL)
         WT_RET(__wt_update_alloc(session, NULL, &append, &size, WT_UPDATE_TOMBSTONE));
-    else if (last_committed_upd != NULL && last_committed_upd->start_ts < unpack->start_ts)
```

Comment:
So basically we exit early here and skip appending the original on disk value, theres a question here as to whether we should even bother inserting any of the updates on the chain into the history store and instead remove them at an earlier stage in reconciliation. I think if we did that instead we may be able to guarantee ordered timestamps here?

Reply:
For more info see the jira.

Reply:
It's weird a modify is based on a full update in the future.

Reply:
Can we only allow full update to be out of order?

Reply:
I agree that it's weird that a modify would apply to a value but have a timestamp in the past of it.  Is that actually what's happening?  I couldn't see details in jira.

Regardless, in my related changes recently I've been trying to "fix" timestamps when things are out of order rather than skipping updates or assert.  Here that would mean taking the on-page value but changing its start time pair to match the stop time pair if they would otherwise be out of order.  So I'm fine with this general approach.

Reply:
Apologies, details are in WT-5799, I had the incorrect jira ID on the PR.